src/drivers/amqp/Queue.php

```diff
@@ -109,21 +162,86 @@ public function status($id)
      */
     protected function open()
     {
-        if ($this->channel) return;
-        $this->connection = new AMQPStreamConnection($this->host, $this->port, $this->user, $this->password);
-        $this->channel = $this->connection->channel();
-        $this->channel->queue_declare($this->queueName, false, true, false, false);
-        $this->channel->exchange_declare($this->exchangeName, 'direct', false, true, false);
-        $this->channel->queue_bind($this->queueName, $this->exchangeName);
+        if ($this->context) {
+            return;
+        }
+
+        switch ($this->driver) {
+            case 'lib':
+                $connectionClass = AmqpLibConnectionFactory::class;
+
+                break;
+            case 'ext':
+                $connectionClass = AmqpExtConnectionFactory::class;
+
+                break;
+            case 'bunny':
+                $connectionClass = AmqpBunnyConnectionFactory::class;
+
+                break;
+
+            default:
+                throw new \LogicException(sprintf('The given driver "%s" is not supported. Supported are "%s"', $this->driver, implode('", "', $this->supportedDrivers)));
+        }
+
+        /** @var AmqpConnectionFactory $factory */
+        $factory = new $connectionClass([
+            'host' => $this->host,
+            'port' => $this->port,
+            'user' => $this->user,
+            'pass' => $this->password,
```

Comment:
I just sent https://github.com/yiisoft/yii2-queue/pull/162 to allow the configuration of vhost. You can include in this PR as well, it is a necessary configuration.

Reply:
sure thing